src/commands/force/user/password/generate.ts

```diff
@@ -51,14 +52,21 @@ export class UserPasswordGenerateCommand extends SfdxCommand {
         const org = await Org.create({ connection });
         const user: User = await User.create({ org });
         const password = User.generatePasswordUtf8();
-        const fields: UserFields = await user.retrieve(username);
+        // we only need the Id, so instead of User.retrieve we'll just query
+        // this avoids permission issues if ProfileId is restricted for the user querying for it
+        const result: QueryResult<{ Id: string }> = await connection.query(
```

Comment:
Here's an example of Connection.singleRecordQuery to throw nice errors on the records[0] out of bounds problems (FYI also the multiple records found problem, which shouldn't be possible on this specific query)

https://github.com/salesforcecli/plugin-org/blob/9913d4319da0ddf48563b199f6044d83f68496a7/src/commands/force/org/display.ts#L88